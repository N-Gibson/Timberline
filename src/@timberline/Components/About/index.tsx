import React from 'react'

import Hallway from '@timberline/assets/Venado_Hallway.jpg'

import './About.scss'

interface AboutProps {}

const About: React.FC<AboutProps> = () => {
  return (
    <section className="about_main">
      <div className="about_banner_container">
        <div id="about_banner">
          <h2>About Us</h2>
        </div>
      </div>
      <section className="about">
        <article className="about_article" id="about_article">
          <p className="about_article_text">
            Timberline Inc. was founded in 1995 with the goals of creating
            quality solutions in the housing market. It has evolved into a
            company that can do a bit of everything. We take pride in creative 
            problem solving for dynamic construction and repair needs.
          </p>
          <img
            className="about_main_image"
            src={Hallway}
            alt="hallway in house"
          />
        </article>
        <article className="about_article">
          <p className="about_article_text">
            Timberline Inc. is not just about creating quality solutions to
            problems that arise throughout the duration of the home owning
            timeline, it is also about creating an extended family within the
            community. Compassion, empathy, understanding are just some of the
            values that sit at the very core of Timberline Inc. Steve Gibson is
            a creative problem solver that will work with you to find the
            perfect solution. If you would like to work with him, please view
            some of the items on his project bucket list!
          </p>
        </article>
        <article className="about_article about_article_bucket_list">
          <h2 className="bucket_list_title">Bucket List</h2>
          <ul className="bucket_list_ul">
            <li>Work on a home from the 1800's</li>
            <li>Build a cabin in Alaska</li>
            <li>Design and build a liveable tree house</li>
          </ul>
        </article>
      </section>
    </section>
  )
}

export { About }
