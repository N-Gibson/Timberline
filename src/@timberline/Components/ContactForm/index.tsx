import React, { useState, createRef } from 'react'

import emailjs from 'emailjs-com'
import { v4 as uuidv4 } from 'uuid'
import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator'
import { Button, TextField } from '@material-ui/core'

import notify from '@timberline/notifications'
import vm from '@timberline/formValidation/validationMessages'

import './ContactForm.scss'
import { useStyles } from './style'

const ContactForm = () => {
  const formDefaults = {
    name: '',
    phone: '',
    email: '',
    description: '',
    loading: false,
    descriptionError: false,
  }

  const [form, setForm] = useState(formDefaults)
  const formRef = createRef()
  const classes = useStyles()

  const setFormChange = (key: any) => ({ target: { value } }: any) => {
    setForm((prev) => ({ ...prev, [key]: value }))
  }

  const sendEmail = (e: any) => {
    e.preventDefault()

    const templateParams = { ...form, id: uuidv4() }

    emailjs
      .send(
        process.env.REACT_APP_EMAILJS_SERVICE_ID!,
        process.env.REACT_APP_EMAILJS_TEMPLATE_ID!,
        templateParams,
        process.env.REACT_APP_EMAILJS_USER_ID!,
      )
      .then(() => {
        notify.success('Work inquiry successfully sent!')
      })
      .catch(() => {
        notify.error('There was an error sending your work inquiry.')
      })

    setForm(formDefaults)
  }

  const toggleDescriptionError = () => {
    !form.description
      ? setForm((prev) => ({ ...prev, descriptionError: true }))
      : setForm((prev) => ({ ...prev, descriptionError: false }))
  }

  const submitOnClick = () => {
    //@ts-ignore
    formRef.current.submit()

    toggleDescriptionError()
  }

  return (
    <ValidatorForm
      className="contact_form"
      onSubmit={sendEmail}
      debounceTime={1500}
      ref={formRef as any}
    >
      <TextValidator
        className={classes.formContent}
        name="name"
        validators={['required']}
        errorMessages={[vm.required]}
        value={form.name}
        label="Name"
        onChange={setFormChange('name')}
      />
      <TextValidator
        className={classes.formContent}
        name="phone"
        type="number"
        validators={['required']}
        errorMessages={[vm.required]}
        value={form.phone}
        label="Phone"
        onChange={setFormChange('phone')}
      />
      <TextValidator
        className={classes.formContent}
        name="email"
        validators={['required', 'isEmail']}
        errorMessages={[vm.required, vm.email]}
        value={form.email}
        label="Email"
        onChange={setFormChange('email')}
      />
      <TextField
        label="Description"
        placeholder="Please include a brief description of the work required here:"
        value={form.description}
        multiline
        variant="outlined"
        rows={5}
        required={true}
        error={form.descriptionError}
        onChange={setFormChange('description')}
      />
      <Button
        className={classes.formContent}
        type="submit"
        onClick={submitOnClick}
      >
        Submit
      </Button>
    </ValidatorForm>
  )
}

export { ContactForm }
