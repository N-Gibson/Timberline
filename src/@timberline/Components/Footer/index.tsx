import React from 'react'
import PermContactCalendarIcon from '@material-ui/icons/PermContactCalendar'
import CallIcon from '@material-ui/icons/Call'
import EmailIcon from '@material-ui/icons/Email'

import './Footer.scss'

const Footer = () => {
  return (
    <footer>
      <div className="footer_subset">
        <PermContactCalendarIcon />
        <p className="footer_content">Contact Us</p>
      </div>
      <div className="footer_subset">
        <CallIcon />
        <p className="footer_content">(505)-470-5446</p>
      </div>
      <div className="footer_subset">
        <EmailIcon />
        <p className="footer_content">timberlineincsantafe@gmail.com</p>
      </div>
    </footer>
  )
}

export { Footer }
