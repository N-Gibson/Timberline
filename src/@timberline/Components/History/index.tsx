import React from 'react'
import Hallway from '../../assets/Venado_Hallway.jpg'
import './History.scss'

const History = () => {
  return (
    <section className="history">
      <article className="history_article">
        <p className="history_article_text">
          Timberline Inc. was founded in 1995 with the goals of creating quality
          solutions in the housing market. It has evolved into a company that
          can do a bit of everything. Etc... this narrative will need some work.
        </p>
        <img
          className="history_main_image"
          src={Hallway}
          alt="hallway in house"
        />
      </article>
    </section>
  )
}

export { History }
