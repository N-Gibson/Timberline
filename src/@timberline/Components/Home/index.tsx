import React from 'react'
import ScrollAnimation from 'react-animate-on-scroll'

import purpleAster from '@timberline/assets/Purple_Aster.jpg'
import profile from '@timberline/assets/profile.jpg'
import timberline_text_only from '@timberline/assets/timberline_text_only.png'

import './Home.scss'

interface HomeProps {}

const Home: React.FC<HomeProps> = () => {
  return (
    <section className="home">
      <div className="banner_container">
        <div className="banner">
          <ScrollAnimation animateIn="fadeIn">
            <div className="logo_text_background">
              <img src={timberline_text_only} alt="timberline logo text"></img>
            </div>
          </ScrollAnimation>
        </div>
      </div>
      <section id="home_center_main" className="section_container">
        <article className="article_container">
          <h2 className="article_header">Raise, Repair, Remodel</h2>
          <p className="article_text">
            Timberline Inc. is a locally owned and operated business in Santa
            Fe, New Mexico and surrounding areas since 1992. Specializing in New
            Mexican style builds and remodels, Timberline can do it all. From
            the preservation of original homes dating back to the mid 1800's to
            brand new homes and everything in between! Award winning Timberline
            Inc. can help with repairs, remodels or raising a new project from
            the ground up.
          </p>
        </article>
        <img
          id="rrr_img"
          className="section_img"
          src={purpleAster}
          alt="house in valley"
        />
      </section>
      <section className="section_container" id="home_bottom_main">
        <img
          id="meet_owner_img"
          className="section_img"
          src={profile}
          alt="timberline owner"
        />
        <article className="article_container">
          <h2 className="article_header">Meet the Owner</h2>
          <p className="article_text">
            Stephen Gibson, owner and founder of Timberline Inc. has been
            working in the Santa Fe community for over 35 years and has received
            11 awards in the Santa Fe area Home Builders Association's - A
            Parade of Homes. Steve's passion is crafting quality solutions for
            your construction needs.
          </p>
        </article>
      </section>
    </section>
  )
}

export { Home }
