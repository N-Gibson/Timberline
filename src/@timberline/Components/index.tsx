export * from './About'
export * from './ContactForm'
export * from './ContactUs'
export * from './Footer'
export * from './History'
export * from './Home'
export * from './Layout'
export * from './Nav'
export * from './Projects'
