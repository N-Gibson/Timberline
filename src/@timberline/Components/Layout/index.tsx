import React from 'react'
import { Nav } from '../Nav'
import { Footer } from '../Footer'

interface LayoutProps {
  children: any
  active: string
  props: any
}

const Layout: React.FC<LayoutProps> = ({ children, active, props }) => {
  return (
    <>
      <Nav props={props} active={active} />
      <main>{children}</main>
      <Footer />
    </>
  )
}

export { Layout }
