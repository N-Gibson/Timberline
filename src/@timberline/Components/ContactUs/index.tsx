import React from 'react'
import './ContactUs.scss'
import { Layout } from '../Layout'
import { ContactForm } from '@timberline/Components/ContactForm'

const ContactUs = () => {
  return (
    <Layout active="contact">
      <div className="contactUs_container">
        <section className="contactUs">
          <article>
            <h3>Interested in remodeling your home?</h3>
            <h2>Get in touch with us for a free quote!</h2>
            <p>
              Eiusmod quis adipisicing irure laborum sint excepteur aute laboris
              aute elit incididunt cupidatat nulla proident. Cillum officia
              cillum do excepteur non laboris cupidatat enim tempor est fugiat
              cupidatat anim. Cupidatat labore nulla pariatur qui. Dolor
              adipisicing est consequat et non officia proident minim duis sunt
              qui voluptate.
            </p>
          </article>
          <ContactForm />
        </section>
      </div>
    </Layout>
  )
}

export { ContactUs }
