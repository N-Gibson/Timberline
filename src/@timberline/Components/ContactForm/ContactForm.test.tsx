import React from 'react'
import { shallow } from 'enzyme'
import { ContactForm } from '@timberline/Components'

describe('ContactForm', () => {
  const wrapper = shallow(<ContactForm />)
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
