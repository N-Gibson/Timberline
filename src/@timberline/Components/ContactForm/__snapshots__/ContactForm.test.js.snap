// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactForm should match the snapshot 1`] = `
<form
  className="contact_form"
  onSubmit={[Function]}
>
  <h2>
    Contact Us
  </h2>
  <input
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Phone Number"
    type="number"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="Email"
    type="text"
    value=""
  />
  <label
    htmlFor="description"
  >
    Short Description
  </label>
  <textarea
    name="description"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Submit
  </button>
</form>
`;
